schema/gen/src/lib.rs

```diff
@@ -1,100 +1,63 @@
 //! Iroha schema generation support library. Contains the
 //! `build_schemas` `fn`, which is the function which decides which
 //! types are included in the schema.
-#![allow(
-    clippy::arithmetic_side_effects,
-    clippy::std_instead_of_core,
-    clippy::std_instead_of_alloc
-)]
+#![allow(clippy::arithmetic_side_effects)]
 
 use iroha_data_model::{block::stream::prelude::*, query::error::QueryExecutionFailure};
 use iroha_genesis::RawGenesisBlock;
 use iroha_schema::prelude::*;
 
-macro_rules! schemas {
-    ($($t:ty),* $(,)?) => {{
-        let mut out = MetaMap::new();
-        $(<$t as IntoSchema>::schema(&mut out);)*
-        out
-    }};
-}
-
 /// Builds the schema for the current state of Iroha.
 ///
 /// You should only include the top-level types, because other types
-/// shall be included automatically.
+/// shall be included recursively.
 pub fn build_schemas() -> MetaMap {
-    use iroha_crypto::MerkleTree;
     use iroha_data_model::prelude::*;
 
+    macro_rules! schemas {
+        ($($t:ty),* $(,)?) => {{
+            let mut out = MetaMap::new(); $(
+            <$t as IntoSchema>::update_schema_map(&mut out); )*
+            out
+        }};
+    }
+
     schemas! {
+        // TODO: Should genesis belong to schema? #3284
         RawGenesisBlock,
 
+        QueryExecutionFailure,
         VersionedBlockMessage,
         VersionedBlockSubscriptionRequest,
         VersionedEventMessage,
         VersionedEventSubscriptionRequest,
         VersionedPaginatedQueryResult,
-        VersionedSignedQueryRequest,
-        VersionedSignedTransaction,
-        QueryExecutionFailure,
-
-        RegistrableBox,
-
-        // Even though these schemas are not exchanged between server and client,
-        // they can be useful to the client to generate and validate their hashes
-        MerkleTree<VersionedSignedTransaction>,
```

Comment:
I'd keep this type.

Reply:
I'm the author of that comment. I removed it because:
1. `MerkleTree` is never exchanged between core and client, only `HashOf<MerkleTree>` is
2. if anyone has to construct `MerkleTree` schema would be of little use for them because it's only a `Vec<Node>`